From 9052a8b77a67a21003191f7246001fe65b169881 Mon Sep 17 00:00:00 2001
From: Valentin Caron <valentin.caron@foss.st.com>
Date: Mon, 2 Sep 2024 11:35:42 +0200
Subject: [PATCH 1223/1695] dts: stm32: describe scmi resources for stm32mp21x

Describe SCMI resources under "scmi" node in stm32mp211.dtsi file

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git ce64cb4654f48169a084f0f8cb0007a9c86f99e2 dts: stm32: describe scmi resources for stm32mp21x]
Signed-off-by: Valentin Caron <valentin.caron@foss.st.com>
Change-Id: I9de88e92cd6f369959fa352e6805e982dcb1beb6
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/401858
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 core/arch/arm/dts/stm32mp211.dtsi | 1811 ++++++++++++++++++++++++++++-
 1 file changed, 1781 insertions(+), 30 deletions(-)

diff --git a/core/arch/arm/dts/stm32mp211.dtsi b/core/arch/arm/dts/stm32mp211.dtsi
index 56ab303d1..cd7aae372 100644
--- a/core/arch/arm/dts/stm32mp211.dtsi
+++ b/core/arch/arm/dts/stm32mp211.dtsi
@@ -8,6 +8,7 @@
 #include <dt-bindings/interrupt-controller/arm-gic.h>
 #include <dt-bindings/regulator/st,stm32mp21-regulator.h>
 #include <dt-bindings/reset/st,stm32mp21-rcc.h>
+#include <dt-bindings/scmi/scmi-clock.h>
 #include <dt-bindings/soc/stm32mp21-rifsc.h>
 #include <dt-bindings/soc/stm32mp25-rif.h>
 #include <dt-bindings/soc/stm32mp25-risab.h>
@@ -134,42 +135,1792 @@
 		};
 	};
 
-	scmi_regu: scmi-regu {
-		compatible = "st,scmi-regulator-consumer";
+	scmi: scmi {
+		compatible = "optee,scmi-server";
 		#address-cells = <1>;
 		#size-cells = <0>;
-		scmi-agent-id = <0>;
-		scmi-channel-id = <0>;
 
-		voltd-vddio1 {
-			reg = <VOLTD_SCMI_VDDIO1>;
-			voltd-supply = <&vddio1>;
-		};
-		voltd-vddio2 {
-			reg = <VOLTD_SCMI_VDDIO2>;
-			voltd-supply = <&vddio2>;
-		};
-		voltd-vddio3 {
-			reg = <VOLTD_SCMI_VDDIO3>;
-			voltd-supply = <&vddio3>;
-		};
-		voltd-vdda18adc {
-			reg = <VOLTD_SCMI_ADC>;
-			voltd-supply = <&vdda18adc>;
+		agent_a35_ns: agent@1 {
+			reg = <1>;
+			compatible = "linaro,scmi-optee";
+			scmi-channel-id = <0>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			scmi_clock: protocol@14 {
+				reg = <0x14>;
+
+				clocks {
+					#address-cells = <1>;
+					#size-cells = <0>;
+					#clock-cells = <1>;
+
+					clock@0 {
+						reg = <CK_SCMI_ICN_HS_MCU>;
+						clocks = <&rcc CK_ICN_HS_MCU>;
+						domain-name = "ck_icn_hs_mcu";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@1 {
+						reg = <CK_SCMI_ICN_SDMMC>;
+						clocks = <&rcc CK_ICN_SDMMC>;
+						domain-name = "ck_icn_sdmmc";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@2 {
+						reg = <CK_SCMI_ICN_DDR>;
+						clocks = <&rcc CK_ICN_DDR>;
+						domain-name = "ck_icn_ddr";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@3 {
+						reg = <CK_SCMI_ICN_DISPLAY>;
+						clocks = <&rcc CK_ICN_DISPLAY>;
+						domain-name = "ck_icn_display";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@4 {
+						reg = <CK_SCMI_ICN_HSL>;
+						clocks = <&rcc CK_ICN_HSL>;
+						domain-name = "ck_icn_hsl";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@5 {
+						reg = <CK_SCMI_ICN_NIC>;
+						clocks = <&rcc CK_ICN_NIC>;
+						domain-name = "ck_icn_nic";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@7 {
+						reg = <CK_SCMI_FLEXGEN_07>;
+						clocks = <&rcc CK_FLEXGEN_07>;
+						domain-name = "ck_flexgen_07";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@8 {
+						reg = <CK_SCMI_FLEXGEN_08>;
+						clocks = <&rcc CK_FLEXGEN_08>;
+						domain-name = "ck_flexgen_08";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@9 {
+						reg = <CK_SCMI_FLEXGEN_09>;
+						clocks = <&rcc CK_FLEXGEN_09>;
+						domain-name = "ck_flexgen_09";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@a {
+						reg = <CK_SCMI_FLEXGEN_10>;
+						clocks = <&rcc CK_FLEXGEN_10>;
+						domain-name = "ck_flexgen_10";
+						flags = <(SCMI_CLOCK_DEFAULT_ENABLED |
+							  SCMI_CLOCK_ALLOW_SET_RATE)>;
+					};
+
+					clock@b {
+						reg = <CK_SCMI_FLEXGEN_11>;
+						clocks = <&rcc CK_FLEXGEN_11>;
+						domain-name = "ck_flexgen_11";
+						flags = <(SCMI_CLOCK_DEFAULT_ENABLED |
+							  SCMI_CLOCK_ALLOW_SET_RATE)>;
+					};
+
+					clock@c {
+						reg = <CK_SCMI_FLEXGEN_12>;
+						clocks = <&rcc CK_FLEXGEN_12>;
+						domain-name = "ck_flexgen_12";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@d {
+						reg = <CK_SCMI_FLEXGEN_13>;
+						clocks = <&rcc CK_FLEXGEN_13>;
+						domain-name = "ck_flexgen_13";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@e {
+						reg = <CK_SCMI_FLEXGEN_14>;
+						clocks = <&rcc CK_FLEXGEN_14>;
+						domain-name = "ck_flexgen_14";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@f {
+						reg = <CK_SCMI_FLEXGEN_15>;
+						clocks = <&rcc CK_FLEXGEN_15>;
+						domain-name = "ck_flexgen_15";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@10 {
+						reg = <CK_SCMI_FLEXGEN_16>;
+						clocks = <&rcc CK_FLEXGEN_16>;
+						domain-name = "ck_flexgen_16";
+						flags = <(SCMI_CLOCK_DEFAULT_ENABLED |
+							  SCMI_CLOCK_ALLOW_SET_RATE)>;
+					};
+
+					clock@11 {
+						reg = <CK_SCMI_FLEXGEN_17>;
+						clocks = <&rcc CK_FLEXGEN_17>;
+						domain-name = "ck_flexgen_17";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@12 {
+						reg = <CK_SCMI_FLEXGEN_18>;
+						clocks = <&rcc CK_FLEXGEN_18>;
+						domain-name = "ck_flexgen_18";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@13 {
+						reg = <CK_SCMI_FLEXGEN_19>;
+						clocks = <&rcc CK_FLEXGEN_19>;
+						domain-name = "ck_flexgen_19";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@14 {
+						reg = <CK_SCMI_FLEXGEN_20>;
+						clocks = <&rcc CK_FLEXGEN_20>;
+						domain-name = "ck_flexgen_20";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@15 {
+						reg = <CK_SCMI_FLEXGEN_21>;
+						clocks = <&rcc CK_FLEXGEN_21>;
+						domain-name = "ck_flexgen_21";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@16 {
+						reg = <CK_SCMI_FLEXGEN_22>;
+						clocks = <&rcc CK_FLEXGEN_22>;
+						domain-name = "ck_flexgen_22";
+						flags = <(SCMI_CLOCK_DEFAULT_ENABLED |
+							  SCMI_CLOCK_ALLOW_SET_RATE)>;
+					};
+
+					clock@17 {
+						reg = <CK_SCMI_FLEXGEN_23>;
+						clocks = <&rcc CK_FLEXGEN_23>;
+						domain-name = "ck_flexgen_23";
+						flags = <(SCMI_CLOCK_DEFAULT_ENABLED |
+							  SCMI_CLOCK_ALLOW_SET_RATE)>;
+					};
+
+					clock@18 {
+						reg = <CK_SCMI_FLEXGEN_24>;
+						clocks = <&rcc CK_FLEXGEN_24>;
+						domain-name = "ck_flexgen_24";
+						flags = <(SCMI_CLOCK_DEFAULT_ENABLED |
+							  SCMI_CLOCK_ALLOW_SET_RATE)>;
+					};
+
+					clock@19 {
+						reg = <CK_SCMI_FLEXGEN_25>;
+						clocks = <&rcc CK_FLEXGEN_25>;
+						domain-name = "ck_flexgen_25";
+						flags = <(SCMI_CLOCK_DEFAULT_ENABLED |
+							  SCMI_CLOCK_ALLOW_SET_RATE)>;
+					};
+
+					clock@1a {
+						reg = <CK_SCMI_FLEXGEN_26>;
+						clocks = <&rcc CK_FLEXGEN_26>;
+						domain-name = "ck_flexgen_26";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@1b {
+						reg = <CK_SCMI_FLEXGEN_27>;
+						clocks = <&rcc CK_FLEXGEN_27>;
+						domain-name = "ck_flexgen_27";
+						flags = <(SCMI_CLOCK_DEFAULT_ENABLED |
+							  SCMI_CLOCK_ALLOW_SET_RATE)>;
+					};
+
+					clock@1c {
+						reg = <CK_SCMI_FLEXGEN_28>;
+						clocks = <&rcc CK_FLEXGEN_28>;
+						domain-name = "ck_flexgen_28";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@1d {
+						reg = <CK_SCMI_FLEXGEN_29>;
+						clocks = <&rcc CK_FLEXGEN_29>;
+						domain-name = "ck_flexgen_29";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@1e {
+						reg = <CK_SCMI_FLEXGEN_30>;
+						clocks = <&rcc CK_FLEXGEN_30>;
+						domain-name = "ck_flexgen_30";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@1f {
+						reg = <CK_SCMI_FLEXGEN_31>;
+						clocks = <&rcc CK_FLEXGEN_31>;
+						domain-name = "ck_flexgen_31";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@20 {
+						reg = <CK_SCMI_FLEXGEN_32>;
+						clocks = <&rcc CK_FLEXGEN_32>;
+						domain-name = "ck_flexgen_32";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@21 {
+						reg = <CK_SCMI_FLEXGEN_33>;
+						clocks = <&rcc CK_FLEXGEN_33>;
+						domain-name = "ck_flexgen_33";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@22 {
+						reg = <CK_SCMI_FLEXGEN_34>;
+						clocks = <&rcc CK_FLEXGEN_34>;
+						domain-name = "ck_flexgen_34";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@23 {
+						reg = <CK_SCMI_FLEXGEN_35>;
+						clocks = <&rcc CK_FLEXGEN_35>;
+						domain-name = "ck_flexgen_35";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@24 {
+						reg = <CK_SCMI_FLEXGEN_36>;
+						clocks = <&rcc CK_FLEXGEN_36>;
+						domain-name = "ck_flexgen_36";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@25 {
+						reg = <CK_SCMI_FLEXGEN_37>;
+						clocks = <&rcc CK_FLEXGEN_37>;
+						domain-name = "ck_flexgen_37";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@26 {
+						reg = <CK_SCMI_FLEXGEN_38>;
+						clocks = <&rcc CK_FLEXGEN_38>;
+						domain-name = "ck_flexgen_38";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@27 {
+						reg = <CK_SCMI_FLEXGEN_39>;
+						clocks = <&rcc CK_FLEXGEN_39>;
+						domain-name = "ck_flexgen_39";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@28 {
+						reg = <CK_SCMI_FLEXGEN_40>;
+						clocks = <&rcc CK_FLEXGEN_40>;
+						domain-name = "ck_flexgen_40";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@29 {
+						reg = <CK_SCMI_FLEXGEN_41>;
+						clocks = <&rcc CK_FLEXGEN_41>;
+						domain-name = "ck_flexgen_41";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@2a {
+						reg = <CK_SCMI_FLEXGEN_42>;
+						clocks = <&rcc CK_FLEXGEN_42>;
+						domain-name = "ck_flexgen_42";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@2b {
+						reg = <CK_SCMI_FLEXGEN_43>;
+						clocks = <&rcc CK_FLEXGEN_43>;
+						domain-name = "ck_flexgen_43";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@2c {
+						reg = <CK_SCMI_FLEXGEN_44>;
+						clocks = <&rcc CK_FLEXGEN_44>;
+						domain-name = "ck_flexgen_44";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@2d {
+						reg = <CK_SCMI_FLEXGEN_45>;
+						clocks = <&rcc CK_FLEXGEN_45>;
+						domain-name = "ck_flexgen_45";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@2e {
+						reg = <CK_SCMI_FLEXGEN_46>;
+						clocks = <&rcc CK_FLEXGEN_46>;
+						domain-name = "ck_flexgen_46";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@2f {
+						reg = <CK_SCMI_FLEXGEN_47>;
+						clocks = <&rcc CK_FLEXGEN_47>;
+						domain-name = "ck_flexgen_47";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@30 {
+						reg = <CK_SCMI_FLEXGEN_48>;
+						clocks = <&rcc CK_FLEXGEN_48>;
+						domain-name = "ck_flexgen_48";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@31 {
+						reg = <CK_SCMI_FLEXGEN_49>;
+						clocks = <&rcc CK_FLEXGEN_49>;
+						domain-name = "ck_flexgen_49";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@32 {
+						reg = <CK_SCMI_FLEXGEN_50>;
+						clocks = <&rcc CK_FLEXGEN_50>;
+						domain-name = "ck_flexgen_50";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@33 {
+						reg = <CK_SCMI_FLEXGEN_51>;
+						clocks = <&rcc CK_FLEXGEN_51>;
+						domain-name = "ck_flexgen_51";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@34 {
+						reg = <CK_SCMI_FLEXGEN_52>;
+						clocks = <&rcc CK_FLEXGEN_52>;
+						domain-name = "ck_flexgen_52";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@35 {
+						reg = <CK_SCMI_FLEXGEN_53>;
+						clocks = <&rcc CK_FLEXGEN_53>;
+						domain-name = "ck_flexgen_53";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@36 {
+						reg = <CK_SCMI_FLEXGEN_54>;
+						clocks = <&rcc CK_FLEXGEN_54>;
+						domain-name = "ck_flexgen_54";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@37 {
+						reg = <CK_SCMI_FLEXGEN_55>;
+						clocks = <&rcc CK_FLEXGEN_55>;
+						domain-name = "ck_flexgen_55";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@38 {
+						reg = <CK_SCMI_FLEXGEN_56>;
+						clocks = <&rcc CK_FLEXGEN_56>;
+						domain-name = "ck_flexgen_56";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@39 {
+						reg = <CK_SCMI_FLEXGEN_57>;
+						clocks = <&rcc CK_FLEXGEN_57>;
+						domain-name = "ck_flexgen_57";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@3a {
+						reg = <CK_SCMI_FLEXGEN_58>;
+						clocks = <&rcc CK_FLEXGEN_58>;
+						domain-name = "ck_flexgen_58";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@3b {
+						reg = <CK_SCMI_FLEXGEN_59>;
+						clocks = <&rcc CK_FLEXGEN_59>;
+						domain-name = "ck_flexgen_59";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@3c {
+						reg = <CK_SCMI_FLEXGEN_60>;
+						clocks = <&rcc CK_FLEXGEN_60>;
+						domain-name = "ck_flexgen_60";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@3d {
+						reg = <CK_SCMI_FLEXGEN_61>;
+						clocks = <&rcc CK_FLEXGEN_61>;
+						domain-name = "ck_flexgen_61";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@3e {
+						reg = <CK_SCMI_FLEXGEN_62>;
+						clocks = <&rcc CK_FLEXGEN_62>;
+						domain-name = "ck_flexgen_62";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@3f {
+						reg = <CK_SCMI_FLEXGEN_63>;
+						clocks = <&rcc CK_FLEXGEN_63>;
+						domain-name = "ck_flexgen_63";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@40 {
+						reg = <CK_SCMI_ICN_LS_MCU>;
+						clocks = <&rcc CK_ICN_LS_MCU>;
+						domain-name = "ck_icn_ls_mcu";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@41 {
+						reg = <CK_SCMI_HSE>;
+						clocks = <&rcc HSE_CK>;
+						domain-name = "hse_ck";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@42 {
+						reg = <CK_SCMI_LSE>;
+						clocks = <&rcc LSE_CK>;
+						domain-name = "lse_ck";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@43 {
+						reg = <CK_SCMI_HSI>;
+						clocks = <&rcc HSI_CK>;
+						domain-name = "hsi_ck";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@44 {
+						reg = <CK_SCMI_LSI>;
+						clocks = <&rcc LSI_CK>;
+						domain-name = "lsi_ck";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@45 {
+						reg = <CK_SCMI_MSI>;
+						clocks = <&rcc MSI_CK>;
+						domain-name = "msi_ck";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@46 {
+						reg = <CK_SCMI_HSE_DIV2>;
+						clocks = <&rcc HSE_DIV2_CK>;
+						domain-name = "hse_div2_ck";
+					};
+
+					clock@47 {
+						reg = <CK_SCMI_CPU1>;
+						clocks = <&rcc CK_CPU1>;
+						domain-name = "ck_cpu1";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@48 {
+						reg = <CK_SCMI_SYSCPU1>;
+						clocks = <&rcc CK_BUS_SYSCPU1>;
+						domain-name = "ck_bus_syscpu1";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@49 {
+						reg = <CK_SCMI_PLL2>;
+						clocks = <&rcc PLL2_CK>;
+						domain-name = "ck_pll2";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@4a {
+						reg = <CK_SCMI_RTC>;
+						clocks = <&rcc CK_BUS_RTC>;
+						domain-name = "ck_bus_rtc";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@4b {
+						reg = <CK_SCMI_RTCCK>;
+						clocks = <&rcc RTC_CK>;
+						domain-name = "rtc_ck";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@4c {
+						reg = <CK_SCMI_ICN_APB1>;
+						clocks = <&rcc CK_ICN_APB1>;
+						domain-name = "ck_icn_apb1";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@4d {
+						reg = <CK_SCMI_ICN_APB2>;
+						clocks = <&rcc CK_ICN_APB2>;
+						domain-name = "ck_icn_apb2";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@4e {
+						reg = <CK_SCMI_ICN_APB3>;
+						clocks = <&rcc CK_ICN_APB3>;
+						domain-name = "ck_icn_apb3";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@4f {
+						reg = <CK_SCMI_ICN_APB4>;
+						clocks = <&rcc CK_ICN_APB4>;
+						domain-name = "ck_icn_apb4";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@50 {
+						reg = <CK_SCMI_ICN_APB5>;
+						clocks = <&rcc CK_ICN_APB5>;
+						domain-name = "ck_icn_apb5";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@51 {
+						reg = <CK_SCMI_ICN_APBDBG>;
+						clocks = <&rcc CK_ICN_APBDBG>;
+						domain-name = "ck_icn_apbdbg";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@52 {
+						reg = <CK_SCMI_TIMG1>;
+						clocks = <&rcc TIMG1_CK>;
+						domain-name = "timg1_ck";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@53 {
+						reg = <CK_SCMI_TIMG2>;
+						clocks = <&rcc TIMG2_CK>;
+						domain-name = "timg2_ck";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@54 {
+						reg = <CK_SCMI_BKPSRAM>;
+						clocks = <&rcc CK_BUS_BKPSRAM>;
+						domain-name = "ck_bus_bkpsram";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@55 {
+						reg = <CK_SCMI_BSEC>;
+						clocks = <&rcc CK_BUS_BSEC>;
+						domain-name = "ck_bus_bsec";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@56 {
+						reg = <CK_SCMI_BUS_ETR>;
+						clocks = <&rcc CK_BUS_ETR>;
+						domain-name = "ck_icn_p_etr";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@57 {
+						reg = <CK_SCMI_FMC>;
+						clocks = <&rcc CK_KER_FMC>;
+						domain-name = "ck_ker_fmc";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@58 {
+						reg = <CK_SCMI_GPIOA>;
+						clocks = <&rcc CK_BUS_GPIOA>;
+						domain-name = "ck_bus_gpioa";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@59 {
+						reg = <CK_SCMI_GPIOB>;
+						clocks = <&rcc CK_BUS_GPIOB>;
+						domain-name = "ck_bus_gpiob";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@5a {
+						reg = <CK_SCMI_GPIOC>;
+						clocks = <&rcc CK_BUS_GPIOC>;
+						domain-name = "ck_bus_gpioc";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@5b {
+						reg = <CK_SCMI_GPIOD>;
+						clocks = <&rcc CK_BUS_GPIOD>;
+						domain-name = "ck_bus_gpiod";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@5c {
+						reg = <CK_SCMI_GPIOE>;
+						clocks = <&rcc CK_BUS_GPIOE>;
+						domain-name = "ck_bus_gpioe";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@5d {
+						reg = <CK_SCMI_GPIOF>;
+						clocks = <&rcc CK_BUS_GPIOF>;
+						domain-name = "ck_bus_gpiof";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@5e {
+						reg = <CK_SCMI_GPIOG>;
+						clocks = <&rcc CK_BUS_GPIOG>;
+						domain-name = "ck_bus_gpiog";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@5f {
+						reg = <CK_SCMI_GPIOH>;
+						clocks = <&rcc CK_BUS_GPIOH>;
+						domain-name = "ck_bus_gpioh";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@60 {
+						reg = <CK_SCMI_GPIOI>;
+						clocks = <&rcc CK_BUS_GPIOI>;
+						domain-name = "ck_bus_gpioi";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@61 {
+						reg = <CK_SCMI_GPIOZ>;
+						clocks = <&rcc CK_BUS_GPIOZ>;
+						domain-name = "ck_bus_gpioz";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@62 {
+						reg = <CK_SCMI_HPDMA1>;
+						clocks = <&rcc CK_BUS_HPDMA1>;
+						domain-name = "ck_bus_hpdma1";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@63 {
+						reg = <CK_SCMI_HPDMA2>;
+						clocks = <&rcc CK_BUS_HPDMA2>;
+						domain-name = "ck_bus_hpdma2";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@64 {
+						reg = <CK_SCMI_HPDMA3>;
+						clocks = <&rcc CK_BUS_HPDMA3>;
+						domain-name = "ck_bus_hpdma3";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@65 {
+						reg = <CK_SCMI_IPCC1>;
+						clocks = <&rcc CK_BUS_IPCC1>;
+						domain-name = "ck_bus_ipcc1";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@66 {
+						reg = <CK_SCMI_RETRAM>;
+						clocks = <&rcc CK_BUS_RETRAM>;
+						domain-name = "ck_bus_retram";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@67 {
+						reg = <CK_SCMI_SRAM1>;
+						clocks = <&rcc CK_BUS_SRAM1>;
+						domain-name = "ck_bus_sram1";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@68 {
+						reg = <CK_SCMI_SYSRAM>;
+						clocks = <&rcc CK_BUS_SYSRAM>;
+						domain-name = "ck_icn_s_sysram";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@69 {
+						reg = <CK_SCMI_OSPI1>;
+						clocks = <&rcc CK_KER_OSPI1>;
+						domain-name = "ck_ker_ospi1";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@6a {
+						reg = <CK_SCMI_TPIU>;
+						clocks = <&rcc CK_KER_TPIU>;
+						domain-name = "ck_ker_tpiu";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@6b {
+						reg = <CK_SCMI_SYSDBG>;
+						clocks = <&rcc CK_SYSDBG>;
+						domain-name = "ck_sys_dbg";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@6c {
+						reg = <CK_SCMI_SYSATB>;
+						clocks = <&rcc CK_BUS_SYSATB>;
+						domain-name = "ck_sys_atb";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@6d {
+						reg = <CK_SCMI_TSDBG>;
+						clocks = <&rcc CK_KER_TSDBG>;
+						domain-name = "ck_ker_tsdbg";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@6e {
+						reg = <CK_SCMI_BUS_STM>;
+						clocks = <&rcc CK_BUS_STM>;
+						domain-name = "ck_icn_p_stm";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@6f {
+						reg = <CK_SCMI_KER_STM>;
+						clocks = <&rcc CK_KER_STM>;
+						domain-name = "ck_icn_s_stm";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@70 {
+						reg = <CK_SCMI_KER_ETR>;
+						clocks = <&rcc CK_KER_ETR>;
+						domain-name = "ck_icn_m_etr";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+				};
+			};
+
+			scmi_reset: protocol@16 {
+				reg = <0x16>;
+
+				resets {
+					#address-cells = <1>;
+					#size-cells = <0>;
+					#reset-cells = <1>;
+
+					reset@0 {
+						reg = <RST_SCMI_C1_R>;
+						domain-name = "c1";
+						resets = <&rcc C1_R>;
+					};
+
+					reset@1 {
+						reg = <RST_SCMI_C2_R>;
+						domain-name = "c2";
+						resets = <&rcc C2_R>;
+					};
+
+					reset@2 {
+						reg = <RST_SCMI_C1_HOLDBOOT_R>;
+						domain-name = "c1_holdboot";
+						resets = <&rcc C1_HOLDBOOT_R>;
+					};
+
+					reset@3 {
+						reg = <RST_SCMI_C2_HOLDBOOT_R>;
+						domain-name = "c2_holdboot";
+						resets = <&rcc C2_HOLDBOOT_R>;
+					};
+
+					reset@4 {
+						reg = <RST_SCMI_FMC>;
+						domain-name = "fmc";
+						resets = <&rcc FMC_R>;
+					};
+
+					reset@5 {
+						reg = <RST_SCMI_OSPI1>;
+						domain-name = "ospi1";
+						resets = <&rcc OSPI1_R>;
+					};
+
+					reset@6 {
+						reg = <RST_SCMI_OSPI1DLL>;
+						domain-name = "ospi1_ddl";
+						resets = <&rcc OSPI1DLL_R>;
+					};
+				};
+			};
+
+			scmi_regulator: protocol@17 {
+				reg = <0x17>;
+
+				scmi_regu: regulators {
+					#address-cells = <1>;
+					#size-cells = <0>;
+
+					voltd-vddio1 {
+						reg = <VOLTD_SCMI_VDDIO1>;
+						voltd-supply = <&vddio1>;
+					};
+
+					voltd-vddio2 {
+						reg = <VOLTD_SCMI_VDDIO2>;
+						voltd-supply = <&vddio2>;
+					};
+
+					voltd-vddio3 {
+						reg = <VOLTD_SCMI_VDDIO3>;
+						voltd-supply = <&vddio3>;
+					};
+
+					voltd-vdda18adc {
+						reg = <VOLTD_SCMI_ADC>;
+						voltd-supply = <&vdda18adc>;
+					};
+				};
+			};
 		};
-	};
 
-	scmi_cm33_agent: scmi-cm-agent@0 {
-		compatible = "st,scmi-cortexm-agent";
-		mbox-names = "scmi_cm33";
-		mboxes = <&ipcc1 RIF_IPCC_CPU1_CHANNEL(16)>;
-		scmi-agent-id = <1>;
-		scmi-channel-id = <0>;
+		agent_m33_ns: agent@2 {
+			reg = <2>;
+			compatible = "arm,scmi";
+			scmi-channel-id = <0>;
+			mbox-names = "scmi_cm33";
+			mboxes = <&ipcc1 RIF_IPCC_CPU1_CHANNEL(16)>;
+			shmem = <&ipc_shmem>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			scmi_clock_m33: protocol@14 {
+				reg = <0x14>;
+
+				clocks {
+					#address-cells = <1>;
+					#size-cells = <0>;
+					#clock-cells = <1>;
+
+					clock@0 {
+						reg = <CK_SCMI_ICN_HS_MCU>;
+						clocks = <&rcc CK_ICN_HS_MCU>;
+						domain-name = "ck_icn_hs_mcu";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@1 {
+						reg = <CK_SCMI_ICN_SDMMC>;
+						clocks = <&rcc CK_ICN_SDMMC>;
+						domain-name = "ck_icn_sdmmc";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@2 {
+						reg = <CK_SCMI_ICN_DDR>;
+						clocks = <&rcc CK_ICN_DDR>;
+						domain-name = "ck_icn_ddr";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
 
-		scmi_regu_cm33: scmi-regu-cm33 {
-			compatible = "st,scmi-regulator-consumer";
-			#address-cells = <1>;
-			#size-cells = <0>;
+					clock@3 {
+						reg = <CK_SCMI_ICN_DISPLAY>;
+						clocks = <&rcc CK_ICN_DISPLAY>;
+						domain-name = "ck_icn_display";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@4 {
+						reg = <CK_SCMI_ICN_HSL>;
+						clocks = <&rcc CK_ICN_HSL>;
+						domain-name = "ck_icn_hsl";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@5 {
+						reg = <CK_SCMI_ICN_NIC>;
+						clocks = <&rcc CK_ICN_NIC>;
+						domain-name = "ck_icn_nic";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@7 {
+						reg = <CK_SCMI_FLEXGEN_07>;
+						clocks = <&rcc CK_FLEXGEN_07>;
+						domain-name = "ck_flexgen_07";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@8 {
+						reg = <CK_SCMI_FLEXGEN_08>;
+						clocks = <&rcc CK_FLEXGEN_08>;
+						domain-name = "ck_flexgen_08";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@9 {
+						reg = <CK_SCMI_FLEXGEN_09>;
+						clocks = <&rcc CK_FLEXGEN_09>;
+						domain-name = "ck_flexgen_09";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@a {
+						reg = <CK_SCMI_FLEXGEN_10>;
+						clocks = <&rcc CK_FLEXGEN_10>;
+						domain-name = "ck_flexgen_10";
+						flags = <(SCMI_CLOCK_DEFAULT_ENABLED |
+							  SCMI_CLOCK_ALLOW_SET_RATE)>;
+					};
+
+					clock@b {
+						reg = <CK_SCMI_FLEXGEN_11>;
+						clocks = <&rcc CK_FLEXGEN_11>;
+						domain-name = "ck_flexgen_11";
+						flags = <(SCMI_CLOCK_DEFAULT_ENABLED |
+							  SCMI_CLOCK_ALLOW_SET_RATE)>;
+					};
+
+					clock@c {
+						reg = <CK_SCMI_FLEXGEN_12>;
+						clocks = <&rcc CK_FLEXGEN_12>;
+						domain-name = "ck_flexgen_12";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@d {
+						reg = <CK_SCMI_FLEXGEN_13>;
+						clocks = <&rcc CK_FLEXGEN_13>;
+						domain-name = "ck_flexgen_13";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@e {
+						reg = <CK_SCMI_FLEXGEN_14>;
+						clocks = <&rcc CK_FLEXGEN_14>;
+						domain-name = "ck_flexgen_14";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@f {
+						reg = <CK_SCMI_FLEXGEN_15>;
+						clocks = <&rcc CK_FLEXGEN_15>;
+						domain-name = "ck_flexgen_15";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@10 {
+						reg = <CK_SCMI_FLEXGEN_16>;
+						clocks = <&rcc CK_FLEXGEN_16>;
+						domain-name = "ck_flexgen_16";
+						flags = <(SCMI_CLOCK_DEFAULT_ENABLED |
+							  SCMI_CLOCK_ALLOW_SET_RATE)>;
+					};
+
+					clock@11 {
+						reg = <CK_SCMI_FLEXGEN_17>;
+						clocks = <&rcc CK_FLEXGEN_17>;
+						domain-name = "ck_flexgen_17";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@12 {
+						reg = <CK_SCMI_FLEXGEN_18>;
+						clocks = <&rcc CK_FLEXGEN_18>;
+						domain-name = "ck_flexgen_18";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@13 {
+						reg = <CK_SCMI_FLEXGEN_19>;
+						clocks = <&rcc CK_FLEXGEN_19>;
+						domain-name = "ck_flexgen_19";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@14 {
+						reg = <CK_SCMI_FLEXGEN_20>;
+						clocks = <&rcc CK_FLEXGEN_20>;
+						domain-name = "ck_flexgen_20";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@15 {
+						reg = <CK_SCMI_FLEXGEN_21>;
+						clocks = <&rcc CK_FLEXGEN_21>;
+						domain-name = "ck_flexgen_21";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@16 {
+						reg = <CK_SCMI_FLEXGEN_22>;
+						clocks = <&rcc CK_FLEXGEN_22>;
+						domain-name = "ck_flexgen_22";
+						flags = <(SCMI_CLOCK_DEFAULT_ENABLED |
+							  SCMI_CLOCK_ALLOW_SET_RATE)>;
+					};
+
+					clock@17 {
+						reg = <CK_SCMI_FLEXGEN_23>;
+						clocks = <&rcc CK_FLEXGEN_23>;
+						domain-name = "ck_flexgen_23";
+						flags = <(SCMI_CLOCK_DEFAULT_ENABLED |
+							  SCMI_CLOCK_ALLOW_SET_RATE)>;
+					};
+
+					clock@18 {
+						reg = <CK_SCMI_FLEXGEN_24>;
+						clocks = <&rcc CK_FLEXGEN_24>;
+						domain-name = "ck_flexgen_24";
+						flags = <(SCMI_CLOCK_DEFAULT_ENABLED |
+							  SCMI_CLOCK_ALLOW_SET_RATE)>;
+					};
+
+					clock@19 {
+						reg = <CK_SCMI_FLEXGEN_25>;
+						clocks = <&rcc CK_FLEXGEN_25>;
+						domain-name = "ck_flexgen_25";
+						flags = <(SCMI_CLOCK_DEFAULT_ENABLED |
+							  SCMI_CLOCK_ALLOW_SET_RATE)>;
+					};
+
+					clock@1a {
+						reg = <CK_SCMI_FLEXGEN_26>;
+						clocks = <&rcc CK_FLEXGEN_26>;
+						domain-name = "ck_flexgen_26";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@1b {
+						reg = <CK_SCMI_FLEXGEN_27>;
+						clocks = <&rcc CK_FLEXGEN_27>;
+						domain-name = "ck_flexgen_27";
+						flags = <(SCMI_CLOCK_DEFAULT_ENABLED |
+							  SCMI_CLOCK_ALLOW_SET_RATE)>;
+					};
+
+					clock@1c {
+						reg = <CK_SCMI_FLEXGEN_28>;
+						clocks = <&rcc CK_FLEXGEN_28>;
+						domain-name = "ck_flexgen_28";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@1d {
+						reg = <CK_SCMI_FLEXGEN_29>;
+						clocks = <&rcc CK_FLEXGEN_29>;
+						domain-name = "ck_flexgen_29";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@1e {
+						reg = <CK_SCMI_FLEXGEN_30>;
+						clocks = <&rcc CK_FLEXGEN_30>;
+						domain-name = "ck_flexgen_30";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@1f {
+						reg = <CK_SCMI_FLEXGEN_31>;
+						clocks = <&rcc CK_FLEXGEN_31>;
+						domain-name = "ck_flexgen_31";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@20 {
+						reg = <CK_SCMI_FLEXGEN_32>;
+						clocks = <&rcc CK_FLEXGEN_32>;
+						domain-name = "ck_flexgen_32";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@21 {
+						reg = <CK_SCMI_FLEXGEN_33>;
+						clocks = <&rcc CK_FLEXGEN_33>;
+						domain-name = "ck_flexgen_33";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@22 {
+						reg = <CK_SCMI_FLEXGEN_34>;
+						clocks = <&rcc CK_FLEXGEN_34>;
+						domain-name = "ck_flexgen_34";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@23 {
+						reg = <CK_SCMI_FLEXGEN_35>;
+						clocks = <&rcc CK_FLEXGEN_35>;
+						domain-name = "ck_flexgen_35";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@24 {
+						reg = <CK_SCMI_FLEXGEN_36>;
+						clocks = <&rcc CK_FLEXGEN_36>;
+						domain-name = "ck_flexgen_36";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@25 {
+						reg = <CK_SCMI_FLEXGEN_37>;
+						clocks = <&rcc CK_FLEXGEN_37>;
+						domain-name = "ck_flexgen_37";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@26 {
+						reg = <CK_SCMI_FLEXGEN_38>;
+						clocks = <&rcc CK_FLEXGEN_38>;
+						domain-name = "ck_flexgen_38";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@27 {
+						reg = <CK_SCMI_FLEXGEN_39>;
+						clocks = <&rcc CK_FLEXGEN_39>;
+						domain-name = "ck_flexgen_39";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@28 {
+						reg = <CK_SCMI_FLEXGEN_40>;
+						clocks = <&rcc CK_FLEXGEN_40>;
+						domain-name = "ck_flexgen_40";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@29 {
+						reg = <CK_SCMI_FLEXGEN_41>;
+						clocks = <&rcc CK_FLEXGEN_41>;
+						domain-name = "ck_flexgen_41";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@2a {
+						reg = <CK_SCMI_FLEXGEN_42>;
+						clocks = <&rcc CK_FLEXGEN_42>;
+						domain-name = "ck_flexgen_42";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@2b {
+						reg = <CK_SCMI_FLEXGEN_43>;
+						clocks = <&rcc CK_FLEXGEN_43>;
+						domain-name = "ck_flexgen_43";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@2c {
+						reg = <CK_SCMI_FLEXGEN_44>;
+						clocks = <&rcc CK_FLEXGEN_44>;
+						domain-name = "ck_flexgen_44";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@2d {
+						reg = <CK_SCMI_FLEXGEN_45>;
+						clocks = <&rcc CK_FLEXGEN_45>;
+						domain-name = "ck_flexgen_45";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@2e {
+						reg = <CK_SCMI_FLEXGEN_46>;
+						clocks = <&rcc CK_FLEXGEN_46>;
+						domain-name = "ck_flexgen_46";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@2f {
+						reg = <CK_SCMI_FLEXGEN_47>;
+						clocks = <&rcc CK_FLEXGEN_47>;
+						domain-name = "ck_flexgen_47";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@30 {
+						reg = <CK_SCMI_FLEXGEN_48>;
+						clocks = <&rcc CK_FLEXGEN_48>;
+						domain-name = "ck_flexgen_48";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@31 {
+						reg = <CK_SCMI_FLEXGEN_49>;
+						clocks = <&rcc CK_FLEXGEN_49>;
+						domain-name = "ck_flexgen_49";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@32 {
+						reg = <CK_SCMI_FLEXGEN_50>;
+						clocks = <&rcc CK_FLEXGEN_50>;
+						domain-name = "ck_flexgen_50";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@33 {
+						reg = <CK_SCMI_FLEXGEN_51>;
+						clocks = <&rcc CK_FLEXGEN_51>;
+						domain-name = "ck_flexgen_51";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@34 {
+						reg = <CK_SCMI_FLEXGEN_52>;
+						clocks = <&rcc CK_FLEXGEN_52>;
+						domain-name = "ck_flexgen_52";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@35 {
+						reg = <CK_SCMI_FLEXGEN_53>;
+						clocks = <&rcc CK_FLEXGEN_53>;
+						domain-name = "ck_flexgen_53";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@36 {
+						reg = <CK_SCMI_FLEXGEN_54>;
+						clocks = <&rcc CK_FLEXGEN_54>;
+						domain-name = "ck_flexgen_54";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@37 {
+						reg = <CK_SCMI_FLEXGEN_55>;
+						clocks = <&rcc CK_FLEXGEN_55>;
+						domain-name = "ck_flexgen_55";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@38 {
+						reg = <CK_SCMI_FLEXGEN_56>;
+						clocks = <&rcc CK_FLEXGEN_56>;
+						domain-name = "ck_flexgen_56";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@39 {
+						reg = <CK_SCMI_FLEXGEN_57>;
+						clocks = <&rcc CK_FLEXGEN_57>;
+						domain-name = "ck_flexgen_57";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@3a {
+						reg = <CK_SCMI_FLEXGEN_58>;
+						clocks = <&rcc CK_FLEXGEN_58>;
+						domain-name = "ck_flexgen_58";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@3b {
+						reg = <CK_SCMI_FLEXGEN_59>;
+						clocks = <&rcc CK_FLEXGEN_59>;
+						domain-name = "ck_flexgen_59";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@3c {
+						reg = <CK_SCMI_FLEXGEN_60>;
+						clocks = <&rcc CK_FLEXGEN_60>;
+						domain-name = "ck_flexgen_60";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@3d {
+						reg = <CK_SCMI_FLEXGEN_61>;
+						clocks = <&rcc CK_FLEXGEN_61>;
+						domain-name = "ck_flexgen_61";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@3e {
+						reg = <CK_SCMI_FLEXGEN_62>;
+						clocks = <&rcc CK_FLEXGEN_62>;
+						domain-name = "ck_flexgen_62";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@3f {
+						reg = <CK_SCMI_FLEXGEN_63>;
+						clocks = <&rcc CK_FLEXGEN_63>;
+						domain-name = "ck_flexgen_63";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@40 {
+						reg = <CK_SCMI_ICN_LS_MCU>;
+						clocks = <&rcc CK_ICN_LS_MCU>;
+						domain-name = "ck_icn_ls_mcu";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@41 {
+						reg = <CK_SCMI_HSE>;
+						clocks = <&rcc HSE_CK>;
+						domain-name = "hse_ck";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@42 {
+						reg = <CK_SCMI_LSE>;
+						clocks = <&rcc LSE_CK>;
+						domain-name = "lse_ck";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@43 {
+						reg = <CK_SCMI_HSI>;
+						clocks = <&rcc HSI_CK>;
+						domain-name = "hsi_ck";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@44 {
+						reg = <CK_SCMI_LSI>;
+						clocks = <&rcc LSI_CK>;
+						domain-name = "lsi_ck";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@45 {
+						reg = <CK_SCMI_MSI>;
+						clocks = <&rcc MSI_CK>;
+						domain-name = "msi_ck";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@46 {
+						reg = <CK_SCMI_HSE_DIV2>;
+						clocks = <&rcc HSE_DIV2_CK>;
+						domain-name = "hse_div2_ck";
+					};
+
+					clock@47 {
+						reg = <CK_SCMI_CPU1>;
+						clocks = <&rcc CK_CPU1>;
+						domain-name = "ck_cpu1";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@48 {
+						reg = <CK_SCMI_SYSCPU1>;
+						clocks = <&rcc CK_BUS_SYSCPU1>;
+						domain-name = "ck_bus_syscpu1";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@49 {
+						reg = <CK_SCMI_PLL2>;
+						clocks = <&rcc PLL2_CK>;
+						domain-name = "ck_pll2";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@4a {
+						reg = <CK_SCMI_RTC>;
+						clocks = <&rcc CK_BUS_RTC>;
+						domain-name = "ck_bus_rtc";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@4b {
+						reg = <CK_SCMI_RTCCK>;
+						clocks = <&rcc RTC_CK>;
+						domain-name = "rtc_ck";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@4c {
+						reg = <CK_SCMI_ICN_APB1>;
+						clocks = <&rcc CK_ICN_APB1>;
+						domain-name = "ck_icn_apb1";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@4d {
+						reg = <CK_SCMI_ICN_APB2>;
+						clocks = <&rcc CK_ICN_APB2>;
+						domain-name = "ck_icn_apb2";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@4e {
+						reg = <CK_SCMI_ICN_APB3>;
+						clocks = <&rcc CK_ICN_APB3>;
+						domain-name = "ck_icn_apb3";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@4f {
+						reg = <CK_SCMI_ICN_APB4>;
+						clocks = <&rcc CK_ICN_APB4>;
+						domain-name = "ck_icn_apb4";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@50 {
+						reg = <CK_SCMI_ICN_APB5>;
+						clocks = <&rcc CK_ICN_APB5>;
+						domain-name = "ck_icn_apb5";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@51 {
+						reg = <CK_SCMI_ICN_APBDBG>;
+						clocks = <&rcc CK_ICN_APBDBG>;
+						domain-name = "ck_icn_apbdbg";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@52 {
+						reg = <CK_SCMI_TIMG1>;
+						clocks = <&rcc TIMG1_CK>;
+						domain-name = "timg1_ck";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@53 {
+						reg = <CK_SCMI_TIMG2>;
+						clocks = <&rcc TIMG2_CK>;
+						domain-name = "timg2_ck";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@54 {
+						reg = <CK_SCMI_BKPSRAM>;
+						clocks = <&rcc CK_BUS_BKPSRAM>;
+						domain-name = "ck_bus_bkpsram";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@55 {
+						reg = <CK_SCMI_BSEC>;
+						clocks = <&rcc CK_BUS_BSEC>;
+						domain-name = "ck_bus_bsec";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@56 {
+						reg = <CK_SCMI_BUS_ETR>;
+						clocks = <&rcc CK_BUS_ETR>;
+						domain-name = "ck_icn_p_etr";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@57 {
+						reg = <CK_SCMI_FMC>;
+						clocks = <&rcc CK_KER_FMC>;
+						domain-name = "ck_ker_fmc";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@58 {
+						reg = <CK_SCMI_GPIOA>;
+						clocks = <&rcc CK_BUS_GPIOA>;
+						domain-name = "ck_bus_gpioa";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@59 {
+						reg = <CK_SCMI_GPIOB>;
+						clocks = <&rcc CK_BUS_GPIOB>;
+						domain-name = "ck_bus_gpiob";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@5a {
+						reg = <CK_SCMI_GPIOC>;
+						clocks = <&rcc CK_BUS_GPIOC>;
+						domain-name = "ck_bus_gpioc";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@5b {
+						reg = <CK_SCMI_GPIOD>;
+						clocks = <&rcc CK_BUS_GPIOD>;
+						domain-name = "ck_bus_gpiod";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@5c {
+						reg = <CK_SCMI_GPIOE>;
+						clocks = <&rcc CK_BUS_GPIOE>;
+						domain-name = "ck_bus_gpioe";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@5d {
+						reg = <CK_SCMI_GPIOF>;
+						clocks = <&rcc CK_BUS_GPIOF>;
+						domain-name = "ck_bus_gpiof";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@5e {
+						reg = <CK_SCMI_GPIOG>;
+						clocks = <&rcc CK_BUS_GPIOG>;
+						domain-name = "ck_bus_gpiog";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@5f {
+						reg = <CK_SCMI_GPIOH>;
+						clocks = <&rcc CK_BUS_GPIOH>;
+						domain-name = "ck_bus_gpioh";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@60 {
+						reg = <CK_SCMI_GPIOI>;
+						clocks = <&rcc CK_BUS_GPIOI>;
+						domain-name = "ck_bus_gpioi";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@61 {
+						reg = <CK_SCMI_GPIOZ>;
+						clocks = <&rcc CK_BUS_GPIOZ>;
+						domain-name = "ck_bus_gpioz";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@62 {
+						reg = <CK_SCMI_HPDMA1>;
+						clocks = <&rcc CK_BUS_HPDMA1>;
+						domain-name = "ck_bus_hpdma1";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@63 {
+						reg = <CK_SCMI_HPDMA2>;
+						clocks = <&rcc CK_BUS_HPDMA2>;
+						domain-name = "ck_bus_hpdma2";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@64 {
+						reg = <CK_SCMI_HPDMA3>;
+						clocks = <&rcc CK_BUS_HPDMA3>;
+						domain-name = "ck_bus_hpdma3";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@65 {
+						reg = <CK_SCMI_IPCC1>;
+						clocks = <&rcc CK_BUS_IPCC1>;
+						domain-name = "ck_bus_ipcc1";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@66 {
+						reg = <CK_SCMI_RETRAM>;
+						clocks = <&rcc CK_BUS_RETRAM>;
+						domain-name = "ck_bus_retram";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@67 {
+						reg = <CK_SCMI_SRAM1>;
+						clocks = <&rcc CK_BUS_SRAM1>;
+						domain-name = "ck_bus_sram1";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@68 {
+						reg = <CK_SCMI_SYSRAM>;
+						clocks = <&rcc CK_BUS_SYSRAM>;
+						domain-name = "ck_icn_s_sysram";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@69 {
+						reg = <CK_SCMI_OSPI1>;
+						clocks = <&rcc CK_KER_OSPI1>;
+						domain-name = "ck_ker_ospi1";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@6a {
+						reg = <CK_SCMI_TPIU>;
+						clocks = <&rcc CK_KER_TPIU>;
+						domain-name = "ck_ker_tpiu";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@6b {
+						reg = <CK_SCMI_SYSDBG>;
+						clocks = <&rcc CK_SYSDBG>;
+						domain-name = "ck_sys_dbg";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@6c {
+						reg = <CK_SCMI_SYSATB>;
+						clocks = <&rcc CK_BUS_SYSATB>;
+						domain-name = "ck_sys_atb";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@6d {
+						reg = <CK_SCMI_TSDBG>;
+						clocks = <&rcc CK_KER_TSDBG>;
+						domain-name = "ck_ker_tsdbg";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@6e {
+						reg = <CK_SCMI_BUS_STM>;
+						clocks = <&rcc CK_BUS_STM>;
+						domain-name = "ck_icn_p_stm";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@6f {
+						reg = <CK_SCMI_KER_STM>;
+						clocks = <&rcc CK_KER_STM>;
+						domain-name = "ck_icn_s_stm";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@70 {
+						reg = <CK_SCMI_KER_ETR>;
+						clocks = <&rcc CK_KER_ETR>;
+						domain-name = "ck_icn_m_etr";
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+				};
+			};
+
+			scmi_reset_m33: protocol@16 {
+				reg = <0x16>;
+
+				resets {
+					#address-cells = <1>;
+					#size-cells = <0>;
+					#reset-cells = <1>;
+
+					reset@0 {
+						reg = <RST_SCMI_C1_R>;
+						domain-name = "c1";
+						resets = <&rcc C1_R>;
+					};
+
+					reset@1 {
+						reg = <RST_SCMI_C2_R>;
+						domain-name = "c2";
+						resets = <&rcc C2_R>;
+					};
+
+					reset@2 {
+						reg = <RST_SCMI_C1_HOLDBOOT_R>;
+						domain-name = "c1_holdboot";
+						resets = <&rcc C1_HOLDBOOT_R>;
+					};
+
+					reset@3 {
+						reg = <RST_SCMI_C2_HOLDBOOT_R>;
+						domain-name = "c2_holdboot";
+						resets = <&rcc C2_HOLDBOOT_R>;
+					};
+
+					reset@4 {
+						reg = <RST_SCMI_FMC>;
+						domain-name = "fmc";
+						resets = <&rcc FMC_R>;
+					};
+
+					reset@5 {
+						reg = <RST_SCMI_OSPI1>;
+						domain-name = "ospi1";
+						resets = <&rcc OSPI1_R>;
+					};
+
+					reset@6 {
+						reg = <RST_SCMI_OSPI1DLL>;
+						domain-name = "ospi1_ddl";
+						resets = <&rcc OSPI1DLL_R>;
+					};
+				};
+			};
+
+			scmi_regulator_m33: protocol@17 {
+				reg = <0x17>;
+
+				regulators {
+					#address-cells = <1>;
+					#size-cells = <0>;
+
+					voltd-vddio1 {
+						reg = <VOLTD_SCMI_VDDIO1>;
+						voltd-supply = <&vddio1>;
+					};
+
+					voltd-vddio2 {
+						reg = <VOLTD_SCMI_VDDIO2>;
+						voltd-supply = <&vddio2>;
+					};
+
+					voltd-vddio3 {
+						reg = <VOLTD_SCMI_VDDIO3>;
+						voltd-supply = <&vddio3>;
+					};
+
+					voltd-vdda18adc {
+						reg = <VOLTD_SCMI_ADC>;
+						voltd-supply = <&vdda18adc>;
+					};
+				};
+			};
 		};
 	};
 
-- 
2.47.2

